### DISCUSSION TITLE
How to exclude getter from being proxified?

### DISCUSSION THREAD

**Discussion by crabvk** [pmndrs/valtio#848] [Q&A] [answered]
I have a class like this:
```typescript
import { proxy, ref } from 'valtio'

class TreeNode {
  expanded = false
  #children = ref({})

  toggle() {
    this.expanded = !this.expanded
  }

  getOrCreate(id: string) {
    if (id in this.#children) {
      return this.#children[id]
    } else {
      const node = proxy(new TreeNode())
      return (this.#children[id] = node)
    }
  }

  // Want to make this getter a ref, but how?
  get childrenLength() {
    Object.keys(this.#children).length
  }
}

export const rootNode = proxy(new TreeNode())
```
and want to exclude `childrenLength` getter from being part of the proxy.

**Comment by dai-shi** [comment]
I don't think it's possible or extremely hard.
Valtio internal function can customize `newProxy`, but that's not generally recommended.

**Comment by crabvk** [comment]
Actually, there's no need to exclude a class field or getter from proxy. Because if I don't want my component to react on a field or computed value changes I can just use `proxyObject.field` instead of `snapshotObject.field`. Is it correct?

**Reply by dai-shi** [reply, marked as answer]
Yes.

See: https://github.com/pmndrs/valtio/blob/26caa885050b5c9ac8c80c90e216599936e925f8/docs/how-tos/how-to-avoid-rerenders-manually.mdx
